components/sets/utils.ts

```diff
@@ -0,0 +1,9 @@
+export const createSetPageLink = (queryParams: Record<string, string>) => {
+  const url = new URL(window.location.href.replace(window.location.search, ''))
```

Comment:
Does the base URL need to be dynamic like this?  Could it look something like:
```ts
  const searchParams = new URLSearchParams()
  Object.entries({ tab: 'sets', ...queryParams }).forEach(([key, value]) => {
    searchParams.set(key, value)
  })
  return `/base/page/path?${searchParams.toString()}`
```

Reply:
yeah there's at least one place where we want to navigate to a nested view while maintaining the parent view so this is the safest way of making sure that we don't lose parent view because of param change.